pkg/console/controllers/clidownloads/controller.go

```diff
@@ -0,0 +1,197 @@
+	consoleClient consolev1client.ConsoleCLIDownloadInterface,
+	routev1Client routeclientv1.RoutesGetter,
+	consoleInformer consolev1informers.ConsoleCLIDownloadInformer,
+	recorder events.Recorder,
+) *CliDownloadsSyncController {
+
+	ctrl := &CliDownloadsSyncController{
+		// clients
+		operatorConfigClient:      operatorClient,
+		consoleCliDownloadsClient: consoleClient,
+		routeClient:               routev1Client,
+		recorder:                  recorder,
+		cachesToSync:              nil,
+		queue:                     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "test"),
+	}
+
+	operatorClient.Informer().AddEventHandler(ctrl.newEventHandler())
+	consoleInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		operatorClient.Informer().HasSynced,
+		consoleInformer.Informer().HasSynced,
+	)
+
+	return ctrl
+}
+
+func (c *CliDownloadsSyncController) sync() error {
+	// spec, status, resourcesVerion, err
+	spec, _, _, err := c.operatorConfigClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	// only run if we are managed
+	if !management.IsOperatorManaged(spec.ManagementState) {
+		return nil
+	}
+
+	consoleRoute, err := c.routeClient.Routes(api.TargetNamespace).Get(api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	host := routesub.GetCanonicalHost(consoleRoute)
+	requiredOCCliDownloads := platformBasedCliDownloads(host, "amd64", ocCliDownloadsCustomResourceName)
+
+	_, _, cliDownloadsErr := GetOrCreate(c.consoleCliDownloadsClient, requiredOCCliDownloads)
+	if cliDownloadsErr != nil {
+		return err
+	}
+
+	return nil
+}
+
+func platformBasedCliDownloads(host, arch, cliDownloadsName string) *v1.ConsoleCLIDownload {
+	baseURL := fmt.Sprintf("%s/%s", util.HTTPS(host), arch)
+	getPlatformURL := func(platform, archiveType string) string {
+		return fmt.Sprintf("%s/%s/%s", baseURL, platform, archiveType)
+	}
+
+	return &v1.ConsoleCLIDownload{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cliDownloadsName,
+		},
+		Spec: v1.ConsoleCLIDownloadSpec{
+			Description: `With the OpenShift command line interface, you can create applications and manage OpenShift projects from a terminal.
+
+	The oc binary offers the same capabilities as the kubectl binary, but it is further extended to natively support OpenShift Container Platform features.		
+	`,
+			DisplayName: "oc - OpenShift Command Line Interface (CLI)",
+			Links: []v1.Link{
+				{
+					Href: getPlatformURL(LinuxPlatformKey, "oc.tar"),
+					Text: "Download oc for Linux",
+				},
+				{
+					Href: getPlatformURL(MacPlatformKey, "oc.tar"),
+					Text: "Download oc for Mac",
+				},
+				{
+					Href: getPlatformURL(WindowsPlatformKey, "oc.tar"),
+					Text: "Download oc for Windows",
+				},
+			},
+		},
+	}
+}
+
+func GetOrCreate(consoleClient consolev1client.ConsoleCLIDownloadInterface, required *v1.ConsoleCLIDownload) (*v1.ConsoleCLIDownload, bool, error) {
```

Comment:
Contemplating, do we want a full `Apply` for this?  If it exists, but someone tampers with it, we probably want to stomp it back to the correct values.